DirectX11Proj/Lighting.hlsl

```diff
@@ -0,0 +1,123 @@
+
+struct Light
+{
+	float3 position;
+	float padL0;
+
+	float3 colour;
+	float padL1;
+};
+
+cbuffer MatrixBuffer : register(b0)
+{
+	matrix worldMatrix;
+	matrix viewMatrix;
+	matrix projectionMatrix;
+	float3 gEyePos;
+};
+
+// integers are 32 bit in HLSL
+cbuffer MaterialBuffer : register(b1)
+{
+	int hasDiffuse;
+	int hasSpecular;
+	int hasNormal;
+	int padding0;
+};
+
+cbuffer LightBuffer : register(b2)
+{
+	int amountOfLights;
+	int padLB01;
+	int padLB02;
+	int padLB03;
+
+
+	Light arr[16];
```

Comment:
Lighting.hlsl(36,12): error X3058: array dimensions must be literal scalar expressions